src/Controller/SubmitController.php

```diff
@@ -32,6 +33,36 @@ public function redirectAction(Request $request) : Response
 
         $jwt = $this->get('elife.journal.security.xpub.token_generator')->generate($user, $request->getSession()->remove('journal.submit') ?? false);
 
-        return new RedirectResponse("{$this->getParameter('submit_url')}#{$jwt}");
+        // if a return url is specified, check that its from a trusted host
+        $returnUrl = $request->query->get('return_url', null);
+
+        if (is_null($returnUrl)) {
+            $returnUrl = $this->getParameter('submit_url');
+        } else {
+            $trustedHosts = $this->getParameter('trusted_hosts');
```

Comment:
This isn't right. `trusted_hosts` is for _this_ app (see https://symfony.com/doc/3.4/reference/configuration/framework.html#trusted-hosts), so needs to be separate.

Reply:
added submit_url_redirects as a separate parameter as we still need to preserve old functionality for xpub